redbot/cogs/warnings/warnings.py

```diff
@@ -300,20 +353,37 @@ def __init__(self, bot: Red):
         await member_settings.total_points.set(current_point_count)
 
         await warning_points_add_check(self.config, ctx, user, current_point_count)
-        try:
+        dm = await self.config.guild(ctx.guild).toggle_dm()
+        if dm:
             em = discord.Embed(
                 title=_("Warning from {user}").format(user=ctx.author),
                 description=reason_type["description"],
             )
             em.add_field(name=_("Points"), value=str(reason_type["points"]))
-            await user.send(
-                _("You have received a warning in {guild_name}.").format(
-                    guild_name=ctx.guild.name
-                ),
-                embed=em,
+            with contextlib.suppress(discord.HTTPException):
+                await user.send(
+                    _("You have received a warning in {guild_name}.").format(
+                        guild_name=ctx.guild.name
+                    ),
+                    embed=em,
+                )
+
+        toggle_channel = await self.config.guild(guild).toggle_channel()
+        if toggle_channel:
+            em = discord.Embed(
+                title=_("Warning from {user}").format(user=ctx.author),
+                description=reason_type["description"],
             )
-        except discord.HTTPException:
-            pass
+            em.add_field(name=_("Points"), value=str(reason_type["points"]))
+            warn_channel = self.bot.get_channel(await self.config.guild(guild).warn_channel())
+            if warn_channel:
+                channel = warn_channel
+                await ctx.tick()
+            else:
+                channel = ctx.channel
+            await channel.send(_("{user} has been warned.").format(user=user.mention), embed=em)
```

Comment:
Sinbad pointed out that this needs handling for cases when bot can't send the message in set channel.
Preferred way of doing this would be first checking if bot has permission to send messages in the channel and also catching any HTTP exception that occurs when sending the message.
```suggestion
            if channel.permissions_for(guild.me).send_messages:
                with contextlib.suppress(discord.HTTPException):
                    await channel.send(
                        _("{user} has been warned.").format(user=user.mention), embed=em
                    )
```